simulator/source/Combat_simulator.cpp

```diff
@@ -256,115 +232,33 @@ void Combat_simulator::cout_damage_parse(Combat_simulator::Hit_type hit_type, So
-        double random_var = get_uniform_random(100);
-        int outcome = std::lower_bound(hit_table_yellow_.begin(), hit_table_yellow_.end(), random_var) -
-                          hit_table_yellow_.begin();
-            return {damage * damage_multipliers_yellow_[outcome], Hit_result(outcome)};
+        damage *= armor_reduction_factor_ * (1 + special_stats.damage_mod_physical);
     }
     else
     {
-        simulator_cout("Drawing outcome from OH hit table");
-        double random_var = get_uniform_random(100);
-        int outcome = std::lower_bound(hit_table_white_oh_.begin(), hit_table_white_oh_.end(), random_var) -
-                      hit_table_white_oh_.begin();
-        return {damage * damage_multipliers_white_oh_[outcome], Hit_result(outcome)};
+        damage *= armor_reduction_factor_add * (1 + special_stats.damage_mod_physical);
     }
-}
 
-Combat_simulator::Hit_outcome Combat_simulator::generate_hit(const Weapon_sim& main_hand_weapon, double damage,
-                                                             Combat_simulator::Hit_type hit_type, Socket weapon_hand,
-                                                             const Special_stats& special_stats,
-                                                             Damage_sources& damage_sources, bool boss_target,
-                                                             bool is_overpower, bool can_sweep, bool is_whirlwind, 
-                                                             bool is_melee_spell)
-{
-    Combat_simulator::Hit_outcome hit_outcome;
-    if (weapon_hand == Socket::main_hand)
-    {
-        hit_outcome = generate_hit_mh(damage, hit_type, is_overpower, is_melee_spell);
-        if (boss_target)
-        {
-            hit_outcome.damage *= armor_reduction_factor_ * (1 + special_stats.damage_mod_physical);
-        }
-        else
-        {
-            hit_outcome.damage *= armor_reduction_factor_add * (1 + special_stats.damage_mod_physical);
-        }
-        cout_damage_parse(hit_type, weapon_hand, hit_outcome);
-    }
-    else
-    {
-        hit_outcome = generate_hit_oh(damage, is_whirlwind);
-        if (boss_target)
-        {
-            hit_outcome.damage *= armor_reduction_factor_ * (1 + special_stats.damage_mod_physical);
-        }
-        else
-        {
-            hit_outcome.damage *= armor_reduction_factor_add * (1 + special_stats.damage_mod_physical);
-        }
-        hit_outcome.damage *= dual_wield_damage_factor_;
-        if (!is_whirlwind)
-        {
-            hit_type = Hit_type::white;
-        }
-        cout_damage_parse(hit_type, weapon_hand, hit_outcome);
-    }
+    auto hit_outcome = hit_table.generate_hit(damage);
+
+    cout_damage_parse(weapon, hit_table, hit_outcome);
 
     if (sweeping_strikes_charges_ > 0)
     {
+        // TODO can_sweep is actually either "clone damage" (can't crit; fully dependent on hit_outcome)
```

Comment:
I've seen sweeping strikes critting on classic vanilla, it uses spell crit chance and modifier (x1.5). Although it is pretty minor and doesn't affect much in grand scheme to be worth implementing

Reply:
No more, it seems - I've tested myself, pretty extensively, and looked through a couple of logs. On "Warcraft Logs", you can find https://tbc.wowhead.com/spell=12723/sweeping-strikes ("clone damage"), and https://tbc.wowhead.com/spell=26654/sweeping-strikes ("normalized swing", for execute & whirlwind). Both are marked as "Can't crit".